From a4af5dd2b0449df19271b22a62a03f70598106f0 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.jansa@gmail.com>
Date: Wed, 11 Sep 2013 18:30:08 +0200
Subject: [PATCH] Allow to build only lrelease + lupdate + lconvert

This is useful e.g. when cross compiling with OpenEmbedded where qtbase-native
is built without GUI support (no-png is set) and we still want to build
native lrelease + lupdate + lconvert tools.

Upstream-Status: Pending

Signed-off-by: Martin Jansa <Martin.jansa@gmail.com>
---
 examples/examples.pro     |  4 ++--
 src/designer/src/src.pro  | 16 +++++++++-------
 src/linguist/linguist.pro |  2 +-
 src/src.pro               | 10 ++++++----
 tests/auto/auto.pro       | 22 ++++++++++++----------
 5 files changed, 30 insertions(+), 24 deletions(-)

diff --git a/examples/examples.pro b/examples/examples.pro
index 4955969..8e86419 100644
--- a/examples/examples.pro
+++ b/examples/examples.pro
@@ -1,3 +1,3 @@
 TEMPLATE = subdirs
-qtHaveModule(widgets): SUBDIRS += help designer linguist uitools assistant
-
+!linguistonly:qtHaveModule(widgets): SUBDIRS += help designer assistant
+qtHaveModule(widgets): SUBDIRS += linguist uitools
diff --git a/src/designer/src/src.pro b/src/designer/src/src.pro
index e02ca8e..49527c5 100644
--- a/src/designer/src/src.pro
+++ b/src/designer/src/src.pro
@@ -1,18 +1,20 @@
 TEMPLATE = subdirs
 
-SUBDIRS = \
-    uitools \
-    lib \
-    components \
-    designer
+!linguistonly {
+    SUBDIRS = \
+        lib \
+        components \
+        designer
+}
+SUBDIRS += uitools
 
-contains(QT_CONFIG, shared): SUBDIRS += plugins
+!linguistonly:contains(QT_CONFIG, shared): SUBDIRS += plugins
 
 components.depends = lib
 designer.depends = components
 plugins.depends = lib
 
-qtNomakeTools( \
+!linguistonly:qtNomakeTools( \
     lib \
     components \
     designer \
diff --git a/src/linguist/linguist.pro b/src/linguist/linguist.pro
index 2979a23..663bdd5 100644
--- a/src/linguist/linguist.pro
+++ b/src/linguist/linguist.pro
@@ -3,7 +3,7 @@ SUBDIRS  = \
     lrelease \
     lupdate \
     lconvert
-!no-png:qtHaveModule(widgets): SUBDIRS += linguist
+!linguistonly:!no-png:qtHaveModule(widgets): SUBDIRS += linguist
 
 qtNomakeTools( \
     linguist \
diff --git a/src/src.pro b/src/src.pro
index 866a199..ec08874 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -4,10 +4,12 @@ qtHaveModule(widgets) {
     no-png {
         message("Some graphics-related tools are unavailable without PNG support")
     } else {
-        SUBDIRS = assistant \
+        !linguistonly { 
+            SUBDIRS = assistant \
                   pixeltool \
-                  qtestlib \
-                  designer
+                  qtestlib
+        }
+        SUBDIRS += designer
 #    unix:!mac:!embedded:!qpa:SUBDIRS += qtconfig
 
         linguist.depends = designer
@@ -25,7 +27,7 @@ android {
     SUBDIRS += androiddeployqt
 }
 
-qtHaveModule(dbus): SUBDIRS += qdbus
+!linguistonly:qtHaveModule(dbus): SUBDIRS += qdbus
 
 win32|winrt:SUBDIRS += windeployqt
 
diff --git a/tests/auto/auto.pro b/tests/auto/auto.pro
index eaf440b..1778ad1 100644
--- a/tests/auto/auto.pro
+++ b/tests/auto/auto.pro
@@ -1,14 +1,16 @@
 TEMPLATE=subdirs
-SUBDIRS=\
-    linguist \
-    host.pro \
-    qhelpcontentmodel \
-    qhelpenginecore \
-    qhelpgenerator \
-    qhelpindexmodel \
-    qhelpprojectdata \
-    cmake \
-    installed_cmake
+!linguistonly {
+    SUBDIRS=\
+        linguist \
+        host.pro \
+        qhelpcontentmodel \
+        qhelpenginecore \
+        qhelpgenerator \
+        qhelpindexmodel \
+        qhelpprojectdata \
+        cmake \
+        installed_cmake
+}
 
 installed_cmake.depends = cmake
 
-- 
1.8.4.3

